src/main/java/io/stargate/sgv2/jsonapi/exception/playing/ErrorCode.java

```diff
@@ -0,0 +1,75 @@
+package io.stargate.sgv2.jsonapi.exception.playing;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Interface for any enum that represents an error code to implement.
+ *
+ * <p>This interface is used because multiple ENUM's define the scopes, the interface creates a way
+ * to treat the values from the different ENUM's in a consistent way.
+ *
+ * <p>The interface makes it easy for code to get an instance of the exception the code represents,
+ * built using the templates error information in {@link ErrorTemplate} that is loaded at startup.
+ *
+ * <p>With this interface code creates an instance of the exception by calling any of the {@link
+ * #get()} overloads for example:
+ *
+ * <pre>
+ *   FilterException.Code.MULTIPLE_ID_FILTER.get("variable_name", "value");
+ *   RequestException.Code.FAKE_CODE.get();
+ * </pre>
+ *
+ * @param <T> Type of the {@link APIException} the error code creates.
+ */
+public interface ErrorCode<T extends APIException> {
```

Comment:
Does this type parameter make sense? If it is used, it needs to be carried along the whole call chain. And generally all we care about is that it we create an `APIException`, as all types have same properties and the only reason (I think) we even have sub-classes is to allow possibly catching and re-throwing different types (server vs request).

It seems to me that leaving out `T` would be simpler -- will try to see if that is true, locally.

Reply:
@amorton Removal of this type parameter seems simple enough, was able to do that in couple of minutes (minus test classes), it's all mechanical. I don't think it adds any value.

Reply:
... but I guess maybe I get it. Caller can get instance of specific exception from factory method.
So it can be cleaner.
I'll have to think more about this. Apologies for stream-of-consciousness kind of commentary :)